tests/test_dice_interface/test_dice_random.py

```diff
@@ -0,0 +1,197 @@
+                                               permitted_range=permitted_range)
+
+    # When a query's feature value is not within the permitted range and the feature is not allowed to vary
+    @pytest.mark.parametrize("features_to_vary, permitted_range, feature_weights",
+                             [(['Numerical'], {'Categorical': ['b', 'c']}, "inverse_mad")])
+    def test_invalid_query_instance(self, sample_custom_query_1, features_to_vary, permitted_range, feature_weights):
+        with pytest.raises(ValueError):
+            self.exp.setup(features_to_vary, permitted_range, sample_custom_query_1, feature_weights)
+
+    # # Testing that the counterfactuals are in the desired class
+    @pytest.mark.parametrize("desired_class, desired_range, total_CFs, features_to_vary, permitted_range",
+                             [(1, None, 2, "all", None)])
+    def test_desired_class(self, desired_class, desired_range, sample_custom_query_2, total_CFs, features_to_vary,
+                           permitted_range):
+        features_to_vary = self.exp.setup(features_to_vary, None, sample_custom_query_2, "inverse_mad")
+        ans = self.exp._generate_counterfactuals(query_instance=sample_custom_query_2,
+                                                 features_to_vary=features_to_vary,
+                                                 total_CFs=total_CFs, desired_class=desired_class,
+                                                 desired_range=desired_range, permitted_range=permitted_range)
+        assert all(ans.final_cfs_df[self.exp.data_interface.outcome_name].values == [desired_class] * total_CFs)
+
+    # Testing that the features_to_vary argument actually varies only the features that you wish to vary
+    @pytest.mark.parametrize("desired_class, desired_range, total_CFs, features_to_vary, permitted_range",
+                             [(1, None, 2, ["Numerical"], None)])
+    def test_features_to_vary(self, desired_class, desired_range, sample_custom_query_2, total_CFs, features_to_vary,
+                              permitted_range):
+        features_to_vary = self.exp.setup(features_to_vary, None, sample_custom_query_2, "inverse_mad")
+        ans = self.exp._generate_counterfactuals(query_instance=sample_custom_query_2,
+                                                 features_to_vary=features_to_vary,
+                                                 total_CFs=total_CFs, desired_class=desired_class,
+                                                 desired_range=desired_range, permitted_range=permitted_range)
+
+        for feature in self.exp.data_interface.feature_names:
+            if feature not in features_to_vary:
+                assert all(ans.final_cfs_df[feature].values[i] == sample_custom_query_2[feature].values[0] for i in
+                           range(total_CFs))
+
+    # # Testing that the permitted_range argument actually varies the features only within the permitted_range
```

Comment:
can these commented lines be removed?

Reply:
I left this check commented as dice_random currently doesn't implement permitted_range properly, in the future if that changes, we can un-comment the test

Reply:
okay

Reply:
Can you mark the test skipped and document the reason why this test is skipped or remove the test case altogether? We shouldn't keep commented out code.